samples/pyarrow/append_rows_with_arrow.py

```diff
@@ -0,0 +1,206 @@
+    bigquery.SchemaField("date_col", enums.SqlTypeNames.DATE),
+    bigquery.SchemaField("datetime_col", enums.SqlTypeNames.DATETIME),
+    bigquery.SchemaField("time_col", enums.SqlTypeNames.TIME),
+    bigquery.SchemaField("timestamp_col", enums.SqlTypeNames.TIMESTAMP),
+    bigquery.SchemaField("geography_col", enums.SqlTypeNames.GEOGRAPHY),
+    bigquery.SchemaField(
+        "range_date_col", enums.SqlTypeNames.RANGE, range_element_type="DATE"
+    ),
+    bigquery.SchemaField(
+        "range_datetime_col",
+        enums.SqlTypeNames.RANGE,
+        range_element_type="DATETIME",
+    ),
+    bigquery.SchemaField(
+        "range_timestamp_col",
+        enums.SqlTypeNames.RANGE,
+        range_element_type="TIMESTAMP",
+    ),
+]
+
+PYARROW_SCHEMA = pa.schema(
+    [
+        pa.field("bool_col", pa.bool_()),
+        pa.field("int64_col", pa.int64()),
+        pa.field("float64_col", pa.float64()),
+        pa.field("numeric_col", pa.decimal128(38, scale=9)),
+        pa.field("bignumeric_col", pa.decimal256(76, scale=38)),
+        pa.field("string_col", pa.string()),
+        pa.field("bytes_col", pa.binary()),
+        pa.field("date_col", pa.date32()),
+        pa.field("datetime_col", pa.timestamp("us")),
+        pa.field("time_col", pa.time64("us")),
+        pa.field("timestamp_col", pa.timestamp("us")),
+        pa.field("geography_col", pa.string()),
+        pa.field(
+            "range_date_col",
+            pa.struct([("start", pa.date32()), ("end", pa.date32())]),
+        ),
+        pa.field(
+            "range_datetime_col",
+            pa.struct([("start", pa.timestamp("us")), ("end", pa.timestamp("us"))]),
+        ),
+        pa.field(
+            "range_timestamp_col",
+            pa.struct([("start", pa.timestamp("us")), ("end", pa.timestamp("us"))]),
+        ),
+    ]
+)
+
+
+def bqstorage_write_client():
+    from google.cloud import bigquery_storage_v1
+
+    return bigquery_storage_v1.BigQueryWriteClient()
+
+
+def make_table(project_id, dataset_id, bq_client):
+    table_id = "append_rows_w_arrow_test"
+    table_id_full = f"{project_id}.{dataset_id}.{table_id}"
+    bq_table = bigquery.Table(table_id_full, schema=BQ_SCHEMA)
+    created_table = bq_client.create_table(bq_table)
+
+    return created_table
+
+
+def create_stream(bqstorage_write_client, table):
+    stream_name = f"projects/{table.project}/datasets/{table.dataset_id}/tables/{table.table_id}/_default"
+    request_template = gapic_types.AppendRowsRequest()
+    request_template.write_stream = stream_name
+
+    # Add schema to the template.
+    arrow_data = gapic_types.AppendRowsRequest.ArrowData()
+    arrow_data.writer_schema.serialized_schema = PYARROW_SCHEMA.serialize().to_pybytes()
+    request_template.arrow_rows = arrow_data
+
+    append_rows_stream = AppendRowsStream(
+        bqstorage_write_client,
+        request_template,
+    )
+    return append_rows_stream
+
+
+def generate_write_requests(num_rows=TABLE_LENGTH):
```

Comment:
I wonder if this would be more readable if is was split up?  You're effectively creating the dataframe and then processing it.  Otherwise, if you keep it together it might be worth a docstring explaining the function?

Reply:
Indeed, updated